test/functional/notes_controller_test.rb

```diff
@@ -81,7 +81,7 @@ def teardown
     assert_equal '0.0.0.0', Impression.last.ip_address
     Impression.last.update_attribute('ip_address', '0.0.0.1')
 
-    assert_difference 'note.totalviews', 1 do
+    assert_difference 'note.reload.totalviews', 1 do
```

Comment:
Oh interesting. Can we add a bit of documentation about this usage? I'm not familiar with it! Thanks!

Reply:
Ah! I see your note above:

> Notice that I update the tests to .reload the node before reading values,
because ActiveRecord caches those, which is why tests fail without it.

Does it cache it just within the context of the test? Any other situations where we'd need to flush the cache? Thank you!!!

Reply:
From my experience with Rails, this is a tests-specific hack. The cache is on request level, so it shouldn't affect any production paths. 
In this specific test, we're reading the note, then changing the impressions, then trying to read a value from the note. Since the note is cached (i.e. already read), we have to force it to reload from DB, so that we get the most recent value of `totalviews`.